rfcs/20220925-bounded-dynamism.md

```diff
@@ -0,0 +1,255 @@
+# StableHLO Bounded Dynamism
+
+[StableHLO](https://github.com/openxla/stablehlo) is an operation set that
+expresses ML computations. It has been originally bootstrapped from the [MHLO
+dialect](https://github.com/tensorflow/mlir-hlo#meta-hlo-dialect-mhlo),
+including inheriting the type and some ops. This RFC aims to describe the
+current status and rationale for bounded dynamism constructs in StableHLO.  In
+particular, this RFC doesn’t propose any further changes to the current state.
+
+Bounded dynamism allows programs to represent the maximum runtime size that a
+particular dynamic dimension of a tensor can have. This makes it possible to run
+such programs on hardware that doesn't support dynamic tensors but could support
```

Comment:
It is mostly a limitation of the software stack rather than the hardware when this happens. Now it may be that the hardware makes it difficult and or the resultant program's performance or space requirements get too great for the HW, but it's not an intrinsic HW issue. Perhaps hardware -> platforms.

Reply:
Ah, better to be generic.